Error on heatmap

HI,

I'm trying use a heatmap. I us like this:

`<pf-heatmap id="id" data="data" max-block-size="20"
						show-legend="false" heatmap-color-pattern="heatmapColorPattern"
						thresholds="thresholds" range-on-hover="false"></pf-heatmap>`

On my script.js:

`$http.get('/monitorRest').success(function(data)` {
	$scope.data= data.monitor;
	$scope.total= data.totalCounter;
	$scope.service= data.serviceCounter;
	$scope.warning= data.warningCounter;
	$scope.error= data.errorCounter;
  });`

The error in browser:

`angular.js:14199 TypeError: Cannot read property 'length' of undefined
    at setSizes (angular-patternfly.js:4548)
    at angular.module.component.controller.ctrl.$postLink (angular-patternfly.js:4755)
    at angular.js:9564
    at forEach (angular.js:333)
    at nodeLinkFn (angular.js:9561)
    at angular.js:9886
    at processQueue (angular.js:16696)
    at angular.js:16712
    at Scope.$eval (angular.js:17994)
    at Scope.$digest (angular.js:17808)
(anonymous) @ angular.js:14199
(anonymous) @ angular.js:10707
processQueue @ angular.js:16704
(anonymous) @ angular.js:16712
$eval @ angular.js:17994
$digest @ angular.js:17808
$apply @ angular.js:18102
bootstrapApply @ angular.js:1812
invoke @ angular.js:4771
doBootstrap @ angular.js:1810
bootstrap @ angular.js:1830
angularInit @ angular.js:1715
(anonymous) @ angular.js:32616
mightThrow @ jquery.js:3583
process @ jquery.js:3651
angular.js:14199 Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: []
http://errors.angularjs.org/1.5.11/$rootScope/infdig?p0=10&p1=%5B%5D
    at angular.js:68
    at Scope.$digest (angular.js:17874)
    at Scope.$apply (angular.js:18102)
    at done (angular.js:12082)
    at completeRequest (angular.js:12291)
    at XMLHttpRequest.requestLoaded (angular.js:12219)
(anonymous) @ angular.js:14199
(anonymous) @ angular.js:10707
$apply @ angular.js:18104
done @ angular.js:12082
completeRequest @ angular.js:12291
requestLoaded @ angular.js:12219
angular.js:68 Uncaught Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: []
http://errors.angularjs.org/1.5.11/$rootScope/infdig?p0=10&p1=%5B%5D
    at angular.js:68
    at Scope.$digest (angular.js:17874)
    at Scope.$apply (angular.js:18102)
    at done (angular.js:12082)
    at completeRequest (angular.js:12291)
    at XMLHttpRequest.requestLoaded (angular.js:12219)
(anonymous) @ angular.js:68
$digest @ angular.js:17874
$apply @ angular.js:18102
done @ angular.js:12082
completeRequest @ angular.js:12291
requestLoaded @ angular.js:12219
angular.js:14199 Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: []
http://errors.angularjs.org/1.5.11/$rootScope/infdig?p0=10&p1=%5B%5D
    at angular.js:68
    at Scope.$digest (angular.js:17874)
    at Scope.$apply (angular.js:18102)
    at angular.js:19950
    at completeOutstandingRequest (angular.js:6045)
    at angular.js:6324
(anonymous) @ angular.js:14199
(anonymous) @ angular.js:10707
$apply @ angular.js:18104
(anonymous) @ angular.js:19950
completeOutstandingRequest @ angular.js:6045
(anonymous) @ angular.js:6324
angular.js:68 Uncaught Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: []
http://errors.angularjs.org/1.5.11/$rootScope/infdig?p0=10&p1=%5B%5D
    at angular.js:68
    at Scope.$digest (angular.js:17874)
    at Scope.$apply (angular.js:18102)
    at angular.js:19950
    at completeOutstandingRequest (angular.js:6045)
    at angular.js:6324
(anonymous) @ angular.js:68
$digest @ angular.js:17874
$apply @ angular.js:18102
(anonymous) @ angular.js:19950
completeOutstandingRequest @ angular.js:6045
(anonymous) @ angular.js:6324
angular.js:68 Uncaught Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: []
http://errors.angularjs.org/1.5.11/$rootScope/infdig?p0=10&p1=%5B%5D
    at angular.js:68
    at Scope.$digest (angular.js:17874)
    at angular.js:18033
    at completeOutstandingRequest (angular.js:6045)
    at angular.js:6324
(anonymous) @ angular.js:68
$digest @ angular.js:17874
(anonymous) @ angular.js:18033
completeOutstandingRequest @ angular.js:6045
(anonymous) @ angular.js:6324
`


The heatmap draws ok, but this error appears and the tooltips not work.

Can you help me?

Thanks

--- a/src/charts/heatmap/heatmap.component.js
+++ b/src/charts/heatmap/heatmap.component.js
@@ -307,10 +307,30 @@ angular.module('patternfly.charts').component('pfHeatmap', {
       });
     };
 
-    ctrl.updateAll = function () {
-      // Need to deep watch changes in chart data
-      prevData = angular.copy(ctrl.data);
+    var updateDisplay = function () {
+      setStyles();
 
+      if (ctrl.chartDataAvailable !== false && ctrl.data) {
+        ctrl.loadingDone = true;
+
+        // Allow the style change to take effect to update the container size
+        $timeout(function () {
+          setSizes();
+          redraw();
+        });
+      }
+    };
+
+    var handleDataUpdate = function () {
+      prevData = angular.copy(ctrl.data);
+      updateDisplay();
+    };
+
+    var debounceResize = _.debounce(function () {
+      updateDisplay();
+    }, 250, 500);
+
+    var updateConfig = function () {
       //Allow overriding of defaults
       if (ctrl.maxBlockSize === undefined || isNaN(ctrl.maxBlockSize)) {
         ctrl.maxSize = 64;
@@ -360,46 +380,36 @@ angular.module('patternfly.charts').component('pfHeatmap', {
       }
       ctrl.height = ctrl.height || heightDefault;
       ctrl.showLegend = ctrl.showLegend || (ctrl.showLegend === undefined);
-      ctrl.loadingDone = false;
-
-      angular.element($window).on('resize', function () {
-        setSizes();
-        redraw();
-      });
-
-      ctrl.thisComponent = $element[0].querySelector('.heatmap-pf-svg');
-
-      $timeout(function () {
-        setStyles();
-        setSizes();
-        redraw();
-      });
     };
+
+    ctrl.loadingDone = false;
 
     ctrl.$onChanges = function (changesObj) {
       if (changesObj.chartDataAvailable && !changesObj.chartDataAvailable.isFirstChange()) {
         setStyles();
-      } else {
-        ctrl.updateAll();
-        ctrl.loadingDone = true;
+      } else if (!changesObj.data) {
+        updateConfig();
+        updateDisplay();
       }
     };
 
     ctrl.$doCheck = function () {
       // do a deep compare on chartData and config
       if (!angular.equals(ctrl.data, prevData)) {
-        setStyles();
-        if (ctrl.chartDataAvailable !== false) {
-          setSizes();
-          redraw();
-        }
+        handleDataUpdate();
       }
     };
 
     ctrl.$postLink = function () {
-      setStyles();
-      setSizes();
-      redraw();
+      ctrl.thisComponent = $element[0].querySelector('.heatmap-pf-svg');
+      updateConfig();
+      handleDataUpdate();
+
+      angular.element($window).on('resize', debounceResize);
+    };
+
+    ctrl.$onDestroy = function () {
+      angular.element($window).off('resize', debounceResize);
     };
   }
 });

--- a/test/charts/heatmap/heatmap.spec.js
+++ b/test/charts/heatmap/heatmap.spec.js
@@ -9,14 +9,16 @@ describe('Component: pfHeatmap', function() {
     'charts/heatmap/heatmap-legend.html'
   ));
 
-  beforeEach(inject(function(_$compile_, _$rootScope_) {
+  beforeEach(inject(function(_$compile_, _$rootScope_, _$timeout_) {
     $compile = _$compile_;
     $scope = _$rootScope_;
+    $timeout = _$timeout_;
   }));
 
   var compileChart = function (markup, scope) {
     var el = $compile(markup)(scope);
     scope.$digest();
+    $timeout.flush();
     return angular.element(el);
   };
 

